parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -148,7 +174,7 @@ public void writeTo(DataOutput out) throws IOException {
 
   private static class FromStringBinary extends ByteArrayBackedBinary {
     public FromStringBinary(byte[] value) {
-      super(value);
+      super(value, true);
```

Comment:
Got it. To do that I will have to call getBytes while invoking super, which takes bytes. .getBytes("UTF-8") needs exception handling which can not be done in the super. Let me know if you know of a way to work around this. As you mentioned the class is private, so this is not a very big concern.